lib/discourse_chat/provider/slack/slack_provider.rb

```diff
@@ -94,6 +112,11 @@ def self.send_via_api(post, channel, message)
       channel: message[:channel].gsub('#', ''),
       attachments: message[:attachments].to_json
     }
+    if message.key?(:thread_ts)
+      data[:thread_ts] = message[:thread_ts]
+    elsif match = slack_thread_regex.match(post.cooked)
```

Comment:
I think you want to match on `post.raw` rather than `post.cooked`. Just in case the comment is stripped out by the markdown engine.

Also, should we add a check for `filter == 'thread'` here, so that we don't write slack thread IDs to the database unnecessarily?

Reply:
`post.raw` yes — I had that and changed to cooked when I looked at what was done for other cases. Makes perfect sense to change it back.

However, in the "only if `filter == 'thread'`" case no, because recording the thread means that if the rule is changed later, responses will go to the right place; it's the same as the reason for storing the thread id when posting a topic even if there is no threading rule currently in place.

Reply:
`post.raw` fixed by typing words until the validator was happy with the complexity of the post. :roll_eyes: